cmd/cosign/cli/sign.go

```diff
@@ -259,12 +258,9 @@ func SignCmd(ctx context.Context, ko KeyOpts, annotations map[string]interface{}
 		}
 	}
 
-	remoteOpts := []remote.Option{
-		remote.WithAuthFromKeychain(authn.DefaultKeychain),
-		remote.WithContext(ctx),
-	}
+	remoteOpts := DefaultRegistryClientOpts(ctx)
 
-	var toSign []name.Digest
+	toSign := make([]name.Digest, 0, len(imgs))
```

Comment:
unnecessary, but not worse than what exists. @jonjohnsonjr assures me that the number of recursive images is unbounded